### PR TITLE
ISPN-16581 Prime the HTTP client connection pool before issuing many …

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.redhat.com/browse/ISPN-16581

### CODE REVIEW

File: client/rest-client/src/main/java/org/infinispan/client/rest/impl/jdk/RestRawClientJDK.java [hunk 1]
```diff
@@ -119,6 +120,14 @@ public class RestRawClientJDK implements RestRawClient, AutoCloseable {
       }
       ServerConfiguration server = configuration.servers().get(0);
       baseURL = String.format("%s://%s:%d", ssl.enabled() ? "https" : "http", server.host(), server.port());
+      if (configuration.pingOnCreate()) {
+         try {
+            head("/").toCompletableFuture().get();
+            System.out.println("PINGED /");
```
Comment (wburns) [inline client/rest-client/src/main/java/org/infinispan/client/rest/impl/jdk/RestRawClientJDK.java:126]: We should probably remove this now. Or convert to a Log trace call?

File: server/rest/src/test/java/org/infinispan/rest/search/RestHitCountAccuracyTest.java [hunk 2]
```diff
@@ -111,7 +111,7 @@ public void test() throws Exception {
       assertEquals(body.at("hit_count").asInteger(), ENTRIES);
    }
 
-   private static void writeEntries(int entries, RestCacheClient cacheClient) {
+   private void writeEntries(int entries, RestCacheClient cacheClient) {
```
Comment (wburns) [inline server/rest/src/test/java/org/infinispan/rest/search/RestHitCountAccuracyTest.java:114]: Any reason this was changed? We should prefer static if it can be.

File: server/rest/src/test/java/org/infinispan/rest/search/aggregation/RestQueryAggregationCountTest.java [hunk 3]
```diff
@@ -60,22 +61,16 @@ protected EmbeddedCacheManager createCacheManager() throws Exception {
...
    }
 
    @Override
    protected void teardown() {
+      Util.close(restClient);
       try {
-         restClient.close();
-      } catch (Exception ex) {
-         // ignore it
+         restServer.stop();
```
Comment (wburns) [inline server/rest/src/test/java/org/infinispan/rest/search/aggregation/RestQueryAggregationCountTest.java:71]: Should this still be wrapped in the try/finally?

**Review by wburns — COMMENTED** [review]
Great work, thanks @tristantarrant !

File: server/rest/src/test/java/org/infinispan/rest/search/RestHitCountAccuracyTest.java (hunk 2 above)
Comment (tristantarrant) [reply server/rest/src/test/java/org/infinispan/rest/search/RestHitCountAccuracyTest.java:114]: It's a test. Also I did this temporarily to invoke `AbstractIinfinispanTest.checkFDs()` in multiple places

**Comment by tristantarrant** [comment]
@wburns addressed your comments

**Review by wburns — APPROVED** [review]
Waiting on CI results as it doesn't have a finished run yet.

**Comment by tristantarrant** [comment]
@wburns look at the GH Action result instead. Jenkins CI is in flux.

**Comment by wburns** [comment]
Looks like RestAccessLoggingTest.testRestAccessLog is related as it has the new HEAD request. And I am not sure on RestEndpointRouterTest.shouldRouteToProperRestServerBasedOnPath, but it is possible.

**Comment by tristantarrant** [comment]
I fixed the two issues. Hopefully we can merge this asap and rebase all other prs on top of it

File: quarkus/integration-tests/server/src/test/resources/batch.file [hunk 4]
```diff
@@ -1 +1 @@
-create cache --template=org.infinispan.DIST_SYNC mybatch
\ No newline at end of file
+create cache --file=target/test-classes/dcache.json mybatch
```
Comment (pruivo) [inline quarkus/integration-tests/server/src/test/resources/batch.file:1]: is this change required? I added the templates to the REST endpoint. 
https://github.com/infinispan/infinispan/pull/12871/files#diff-6b434edf0f299b314c0f0f6b046e90ca3653460d5e1787d7d26a4a15862a033eR154

File: server/tests/src/test/java/org/infinispan/server/cli/CliIT.java [hunk 5]
```diff
@@ -70,7 +70,8 @@ public void testCliInteractive() {
          terminal.send("stats");
          terminal.assertContains("required_minimum_number_of_nodes");
          terminal.clear();
-         terminal.send("create cache --template=org.infinispan.DIST_SYNC dcache");
+         terminal.send("create cache --file=" + getCliResource("dtemplate.xml").getPath() + " dtemplate");
```
Comment (pruivo) [inline server/tests/src/test/java/org/infinispan/server/cli/CliIT.java:73]: same question here :)

File: server/tests/src/test/java/org/infinispan/server/functional/rest/RestOperations.java [hunk 6]
```diff
@@ -117,8 +117,7 @@ public void testSSECluster(Protocol protocol) throws Exception {
 
       try (Closeable ignored = client.raw().listen("/rest/v2/container?action=listen", Collections.emptyMap(), sseListener)) {
          assertTrue(sseListener.await(10, TimeUnit.SECONDS));
-
-         assertThat(client.cache("caching-listen").createWithTemplate("org.infinispan.DIST_SYNC")).isOk();
+         assertThat(client.cache("caching-listen").createWithConfiguration(RestEntity.create(MediaType.APPLICATION_JSON, "{\"distributed-cache\":{}}"))).isOk();
```
Comment (pruivo) [inline server/tests/src/test/java/org/infinispan/server/functional/rest/RestOperations.java:120]: another :)

File: quarkus/integration-tests/server/src/test/resources/batch.file (hunk 4 above)
Comment (tristantarrant) [reply quarkus/integration-tests/server/src/test/resources/batch.file:1]: While adding those template names is ok, I'd prefer it if we didn't rely on them in our tests

**Comment by tristantarrant** [comment]
Looking great

**Comment by wburns** [comment]
Integrated into main, thanks @tristantarrant !
